### DISCUSSION TITLE
Template for private docker registery

### DISCUSSION THREAD

**Discussion by hannes09** [coder/coder#8188] [General] [answered]
Hi,

I could not find a template for my use case, so I am asking for help:

**My setup:**
I have a docker image that contains everything i need for my development, including my sdks. The docker image is in a private docker registery. 
I can access the registery only via certificates (*.crt  *pki.crt  *.crt), that my organization  distributes. 
Normaly I install this files in  /usr/local/share/ca-certificates. After that I can use docker pull /push.

I am using the OSS coder v2 version.
Coder is installed on I linux host (Ubunutu 22.04) that I manage.

I want to create a template that pulls my image form my private registery. 
IDE should be VSCode.

I had a look at https://github.com/coder/coder/blob/main/examples/templates/docker/main.tf
but for me its unclear how can I edit this template to include my certificates and also how to include my image.

**Comment by matifali** [comment, marked as answer]
Please check here on how to configure the authentication for private registries
https://registry.terraform.io/providers/kreuzwerker/docker/latest/docs
Feel free to ping me with further questions.

**Reply by hannes09** [reply]
Thanks, works now
